### DISCUSSION TITLE
The getter 'future' isn't defined - n00b alert

### DISCUSSION THREAD

**Discussion by amireldor** [rrousselGit/riverpod#3444] [Q&A] [answered]
Hi, I'm rather new to Flutter/Dart and I think I'm doing something silly. When trying to wait for a `.future` on a provider from another provider, I get a warning that this `.future` getter is not defined.

```dart
import 'package:riverpod_annotation/riverpod_annotation.dart';

part 'auth.g.dart';

// Did I do something wrong here?
@riverpod
Future<String> authProvider(AuthProviderRef ref) async {
  await Future.delayed(const Duration(seconds: 3));
  return 'hello';
}
```

And on another provider:
```dart
import 'package:nice_app/providers/auth.dart';
import 'package:riverpod_annotation/riverpod_annotation.dart';

part 'games.g.dart';

@riverpod
Future<void> gamesProvider(GamesProviderRef ref) async {
  await ref.watch(authProvider.future); // warning attached below
}
```

And I get the following warning:
![image](https://github.com/rrousselGit/riverpod/assets/1746811/74c3c96b-a3b8-4a10-bcd8-95ff39b14e3a)

I am probably missing something very basic but it's almost 2am and my brain is not working. Thanks.

**Comment by charlescyt** [comment, marked as answer]
That's because `authProvider` is not referring to the actual generated `FutureProvider`. It is referring to the function

```dart
@riverpod
Future<String> authProvider(AuthProviderRef ref) async {
  await Future.delayed(const Duration(seconds: 3));
  return 'hello';
}
```

Change the code to `await ref.watch(authProviderProvider.future);` and it will work.
But this is rather confusing having a `ProviderProvider` so generally you wouldn't suffix your provider with provider.

```dart
@riverpod
Future<String> auth(AuthRef ref) async {
  await Future.delayed(const Duration(seconds: 3));
  return 'hello';
}
```

**Reply by amireldor** [reply]
Thanks 🤣🤣 
I was missing the part about the auto-suffix.
